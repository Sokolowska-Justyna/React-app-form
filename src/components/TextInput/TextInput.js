import React from "react";
import "./TextInput.scss";

export const TextInput = props => {
  const { placeholder, name, required, onChange, value } = props;

  return (
    <input
      type="text"
      name={name}
      placeholder={placeholder}
      required={required}
      onChange={onChange}
      value={value}
      className="text-input"
    />
  );
};
