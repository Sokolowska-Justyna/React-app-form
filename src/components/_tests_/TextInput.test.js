import React from 'react';
import { TextInput } from '../TextInput';
import { shallow } from 'enzyme';

describe('TextInput', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<TextInput 
      placeholder="placeholder"
      name="text-input-name"
      onChange={jest.fn()}
    />);
  
    expect(wrapper).toMatchSnapshot();
  });
});