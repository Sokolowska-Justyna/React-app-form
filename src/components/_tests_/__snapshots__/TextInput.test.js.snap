// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    name="text-input-name"
    onChange={[MockFunction]}
    placeholder="placeholder"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "name": "text-input-name",
      "onChange": [MockFunction],
      "placeholder": "placeholder",
      "required": undefined,
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "name": "text-input-name",
        "onChange": [MockFunction],
        "placeholder": "placeholder",
        "required": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
